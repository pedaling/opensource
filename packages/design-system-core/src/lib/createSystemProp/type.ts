import type { Theme } from '../types';

export type InterpolationFn = (props: Record<string, any>) => Record<string, any>;

export type SystemPropConfig = {
  property: string;
  styleProperty?: string;
  scale?: keyof Pick<Theme, 'colors' | 'space' | 'opacity'>;
  disabled?: boolean;
  transform?: (value: any) => Record<string, any> | undefined;
};

export type SystemProp = {
  (props: Record<string, any>): Record<string, any>;
  propName: string;
  disabled: boolean;
};
