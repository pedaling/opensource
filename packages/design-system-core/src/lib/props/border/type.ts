import type { ResponsiveValue } from '../../types';

export type BorderProps = {
  borderWidth?: ResponsiveValue<number>;
  borderStyle?: ResponsiveValue<'solid' | 'none'>;
  borderColor?: ResponsiveValue<string>;
  borderTopWidth?: ResponsiveValue<number>;
  borderTopStyle?: ResponsiveValue<'solid' | 'none'>;
  borderTopColor?: ResponsiveValue<string>;
  borderRightWidth?: ResponsiveValue<number>;
  borderRightStyle?: ResponsiveValue<'solid' | 'none'>;
  borderRightColor?: ResponsiveValue<string>;
  borderBottomWidth?: ResponsiveValue<number>;
  borderBottomStyle?: ResponsiveValue<'solid' | 'none'>;
  borderBottomColor?: ResponsiveValue<string>;
  borderLeftWidth?: ResponsiveValue<number>;
  borderLeftStyle?: ResponsiveValue<'solid' | 'none'>;
  borderLeftColor?: ResponsiveValue<string>;
  borderRadius?: ResponsiveValue<number | string>;
  borderTopLeftRadius?: ResponsiveValue<number | string>;
  borderTopRightRadius?: ResponsiveValue<number | string>;
  borderBottomLeftRadius?: ResponsiveValue<number | string>;
  borderBottomRightRadius?: ResponsiveValue<number | string>;
  outlineWidth?: ResponsiveValue<number>;
  outlineStyle?: ResponsiveValue<'solid' | 'none'>;
  outlineColor?: ResponsiveValue<string>;
  outlineOffset?: ResponsiveValue<number>;
};
