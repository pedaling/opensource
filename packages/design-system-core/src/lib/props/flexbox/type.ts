import type { ResponsiveValue } from '../../types';

export type FlexboxProps = {
  flex?: ResponsiveValue<string | number>;
  flexGrow?: ResponsiveValue<string | number>;
  flexShrink?: ResponsiveValue<string | number>;
  flexBasis?: ResponsiveValue<string | number>;
  flexDirection?: ResponsiveValue<'column' | 'column-reverse' | 'row' | 'row-reverse'>;
  flexWrap?: ResponsiveValue<'nowrap' | 'wrap' | 'wrap-reverse'>;
  alignContent?: ResponsiveValue<
    | 'space-around'
    | 'space-between'
    | 'space-evenly'
    | 'stretch'
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'start'
    | 'baseline'
    | 'normal'
  >;
  alignItems?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'baseline'
    | 'normal'
    | 'stretch'
  >;
  alignSelf?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'auto'
    | 'baseline'
    | 'normal'
    | 'stretch'
  >;
  justifyContent?: ResponsiveValue<
    | 'space-around'
    | 'space-between'
    | 'space-evenly'
    | 'stretch'
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'start'
    | 'left'
    | 'normal'
    | 'right'
  >;
  justifyItems?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'baseline'
    | 'left'
    | 'legacy'
    | 'normal'
    | 'right'
    | 'stretch'
  >;
  justifySelf?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'auto'
    | 'baseline'
    | 'left'
    | 'normal'
    | 'right'
    | 'stretch'
  >;
  order?: ResponsiveValue<number>;
};
