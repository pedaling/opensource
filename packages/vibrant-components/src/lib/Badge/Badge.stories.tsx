import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { Icon } from '@vibrant-ui/icons';
import { Badge } from './Badge';

export default {
  title: 'Badge',
  component: Badge,

  args: {
    kind: 'primary',
    size: 'md',
    pill: true,
    children: 'this is a badge',
    IconComponent: Icon.Pin.Fill,
  },
} as ComponentMeta<typeof Badge>;

export const Basic: ComponentStory<typeof Badge> = props => <Badge {...props} />;
