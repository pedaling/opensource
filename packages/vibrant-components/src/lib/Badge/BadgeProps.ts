import type { ReactElement } from 'react';
import type { TextSystemProps } from '@vibrant-ui/core';
import { type ResponsiveValue, propVariant, withVariation } from '@vibrant-ui/core';
import type { IconComponent, IconProps } from '@vibrant-ui/icons';
import type { Rounded } from '@vibrant-ui/theme';

type BadgeKind =
  | 'error'
  | 'errorWeak'
  | 'inverse'
  | 'neutral'
  | 'primary'
  | 'primaryWeak'
  | 'secondary'
  | 'secondaryWeak'
  | 'success'
  | 'successWeak'
  | 'warning'
  | 'warningWeak';

export type BadgeProps = {
  kind: BadgeKind;
  size: ResponsiveValue<'lg' | 'md' | 'sm'>;
  IconComponent?: IconComponent<IconProps, 'Fill' | 'Regular' | 'Thin'>;
  children: ReactElement | number | string;
  rounded?: ResponsiveValue<Rounded>;
  whiteSpace?: TextSystemProps['whiteSpace'];
};

export const withBadgeVariation = withVariation<BadgeProps>('Badge')(
  propVariant({
    props: [
      {
        name: 'kind',
      },
    ],
    variants: {
      warning: {
        backgroundColor: 'warning',
        color: 'onWarning',
      },
      neutral: {
        backgroundColor: 'surface1',
        color: 'onView1',
      },
      primary: {
        backgroundColor: 'primary',
        color: 'onPrimary',
      },
      secondary: {
        backgroundColor: 'informative',
        color: 'onInformative',
      },
      success: {
        backgroundColor: 'success',
        color: 'onSuccess',
      },
      inverse: {
        backgroundColor: 'inverseBackground',
        color: 'onInverseSurface',
      },
      error: {
        backgroundColor: 'error',
        color: 'onError',
      },
      secondaryWeak: {
        backgroundColor: 'informativeContainer',
        color: 'onInformativeContainer',
      },
      warningWeak: {
        backgroundColor: 'warningContainer',
        color: 'onWarningContainer',
      },
      primaryWeak: {
        backgroundColor: 'primaryContainer',
        color: 'onPrimaryContainer',
      },
      errorWeak: {
        backgroundColor: 'errorContainer',
        color: 'onErrorContainer',
      },
      successWeak: {
        backgroundColor: 'successContainer',
        color: 'onSuccessContainer',
      },
    } as const,
  }),
  propVariant({
    props: [{ name: 'size', responsive: true }],
    variants: {
      sm: {
        px: 6,
        py: 3,
        iconSize: 10,
        bodyLevel: 5,
      },
      md: {
        px: 6,
        py: 4,
        iconSize: 12,
        bodyLevel: 4,
      },
      lg: {
        px: 8,
        py: 6,
        iconSize: 14,
        bodyLevel: 3,
      },
    } as const,
  })
);
