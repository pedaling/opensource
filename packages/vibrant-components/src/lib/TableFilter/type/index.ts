export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
