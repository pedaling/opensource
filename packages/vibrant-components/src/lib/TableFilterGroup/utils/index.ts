import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
