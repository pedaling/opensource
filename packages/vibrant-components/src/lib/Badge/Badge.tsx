import { Box } from '@vibrant-ui/core';
import { Body } from '../Body';
import { withBadgeVariation } from './BadgeProps';

export const Badge = withBadgeVariation(
  ({ testId = 'badge', iconSize, bodyLevel, color, IconComponent, whiteSpace, children, ...rest }) => (
    <Box {...rest} display="inline-flex" data-testid={testId} flexDirection="row" alignItems="center" gap={4}>
      {IconComponent && <IconComponent size={iconSize} fill={color} />}
      <Body level={bodyLevel} weight="medium" color={color} whiteSpace={whiteSpace}>
        {children}
      </Body>
    </Box>
  )
);
