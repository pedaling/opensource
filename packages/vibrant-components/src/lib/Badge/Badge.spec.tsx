import { Icon } from '@vibrant-ui/icons';
import type { ReactRenderer } from '@vibrant-ui/utils/testing-web';
import { createReactRenderer } from '@vibrant-ui/utils/testing-web';
import { Badge } from './Badge';

describe('<Badge />', () => {
  const { render } = createReactRenderer();
  let renderer: ReactRenderer;

  describe('when size is sm and kind is primary with IconComponent', () => {
    beforeEach(async () => {
      renderer = render(
        <Badge size="sm" IconComponent={Icon.Pin.Fill} kind="primary">
          this is a badge
        </Badge>
      );
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });
});
