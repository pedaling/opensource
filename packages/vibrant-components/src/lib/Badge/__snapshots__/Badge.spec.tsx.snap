// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> when size is sm and kind is primary with IconComponent match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: flex-start;
  align-self: flex-start;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  border-radius: 4px;
  background-color: #ff5d00;
  padding-left: 6px;
  padding-right: 6px;
  padding-top: 3px;
  padding-bottom: 3px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 4px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 10px;
  height: 10px;
  fill: #ffffff;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
}

.emotion-3 {
  text-align: left;
  color: #ffffff;
  -webkit-text-fill-color: #ffffff;
  text-fill-color: #ffffff;
  font-size: 0.6875rem;
  line-height: 0.875rem;
  font-weight: 500;
}

<div>
  <div
    class="emotion-0"
    data-testid="badge"
  >
    <svg
      class="emotion-1"
      data-testid="pin-fill"
      viewBox="0 0 24 24"
    >
      <path
        class="emotion-2"
        d="M17.15 2H6.85c-.15 0-.25.1-.25.25v7.1c0 .1 0 .15-.05.25l-3.5 6.5c-.05.05-.05.1-.05.15 0 .15.1.25.25.25h7.5v4.6c0 .1.05.2.1.25l.95 1.55c.05.1.15.1.2.1s.15-.05.2-.1l.95-1.55c.05-.1.1-.15.1-.25v-4.6h7.5c.15 0 .25-.1.25-.25 0-.05 0-.1-.05-.1l-3.5-6.5c-.05-.05-.05-.15-.05-.25V2.25c0-.15-.1-.25-.25-.25"
      />
    </svg>
    <span
      class="emotion-3"
      data-testid="body"
    >
      this is a badge
    </span>
  </div>
</div>
`;
